agent/fw_slim.c

```diff
@@ -81,28 +81,64 @@ NR_PHP_WRAPPER(nr_slim2_route_dispatch) {
 NR_PHP_WRAPPER_END
 
 /*
- * Wrap the \Slim\Route::run method, which is the happy path for Slim routing.
+ * Wrap the Slim 3\Slim\Route::run method
+ * and
+ * Slim 4 Slim\\Routing\\Route::run 
+  * which are the happy paths for Slim 3/4 routing.
  * i.e. The router has succesfully matched the URL and dispatched the request
  * to a route.
  *
  * In this case, `nr_txn_set_path` is called after `NR_PHP_WRAPPER_CALL` with
- * `NR_OK_TO_OVERWRITE` and as this corresponds to calling the wrapped function
- * in func_end no change is needed to ensure OAPI compatibility as it will use
- * the default func_end after callback. This entails that the first wrapped
- * function call of this type gets to name the txn.
+ * `NR_OK_TO_OVERWRITE` before wrapper_call and as this corresponds to calling
+ * the last wrapped function call of this type gets to name the txn therefore
+ * needs a before call for OAPI.
```

Comment:
This comment doesn't seem right - it says two contradicting things: after `NR_PHP_WRAPPER_CALL` and before wrapper_call:

> ... `nr_txn_set_path` is called after `NR_PHP_WRAPPER_CALL` with
`NR_OK_TO_OVERWRITE` before wrapper_call

I think some of it is a left over from before this PR, when `nr_txn_set_path` was actually called *after* `NR_PHP_WRAPPER_CALL`. I think it should be changed to:
> ... `nr_txn_set_path` is called before `NR_PHP_WRAPPER_CALL` with
`NR_OK_TO_OVERWRITE`

Reply:
good catch, thanks!
please see: [3ac4075](https://github.com/newrelic/newrelic-php-agent/pull/917/commits/3ac40759826e4ba64bdc45785549f922c167d83b)